File: javascripts/js/h00.overriding/Rectangle.js
load('Shape.js');
/******* CLASS DEFINITION *******/
var Rectangle = (function () {
  Rectangle.prototype = new Shape(); // use shape for basis of prototype 

  /******* CONSTRUCTOR *******/
  function Rectangle (width, height) { Shape.call(this, width, height); }

  /******* METHODS *******/
  Rectangle.prototype.calculateArea = function () {
    return (this.width * this.height);
  };

  return Rectangle;
})(Shape);

File: javascripts/dart/i00.overriding/Rectangle.dart
library Shape;
import "Shape.dart";

/******* CLASS DECLARATION *******/
class Rectangle extends Shape {
  /******* CONSTRUCTORS *******/
  Rectangle(double width, double height) : super(width, height) { /* Nothing */ }

  /******* METHODS *******/
  double calculateArea() { return (this.width * this.height); }
}
